craft_cli/messages.py

```diff
@@ -780,6 +780,19 @@ def error(self, error: errors.CraftError) -> None:
         self._report_error(error)
         self._stop()
 
+    @_active_guard()
+    def dump_log_contents(self, log_path: pathlib.Path | str, prefix: str = "::") -> None:
+        """Dump the contents of an external log file into the emitter log."""
+        # Open a path object in case we received a string
+        if isinstance(log_path, str):
+            log_path = pathlib.Path(log_path)
+
+        with log_path.open() as log_file:
+            for line in log_file:
+                text = f"{prefix} {line}"
```

Comment:
Alex had another suggestion implicitly (unconsciouly?) on his comment: I don't think we should hardcode a space here between `prefix` and `line`, and the method's default should be `":: "` (with the space). That way consumers can use whatever prefix they want, including no prefix at all with `""`